Incorrect behavior for `pathlib.Path.resolve()` when path does not exist

In the following test suite, the test that does not use the `fs` fixture succeeds while the test that does use the `fs` fixture fails:
```python
import pathlib

import pytest


def test_pathlib_resolve_without_fs():
    path = pathlib.Path('/var/foo/bar/baz')

    assert path.resolve() == path


@pytest.mark.usefixtures('fs')
def test_pathlib_resolve_with_fs():
    path = pathlib.Path('/var/foo/bar/baz')

    assert path.resolve() == path
```
Here is a snippet of the results:
```
===================================================== FAILURES =====================================================
___________________________________________ test_pathlib_resolve_with_fs ___________________________________________

>   ???
E   AssertionError: assert PosixPath('/var') == PosixPath('/var/foo/bar/baz')
E    +  where PosixPath('/var') = <bound method FakePath.resolve of PosixPath('/var/foo/bar/baz')>()
E    +    where <bound method FakePath.resolve of PosixPath('/var/foo/bar/baz')> = PosixPath('/var/foo/bar/baz').resolve
```

From the documentation (https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve):
```
If strict is False, the path is resolved as far as possible and any remainder is appended without checking whether it exists.
```

A [similar bug](https://bugs.python.org/issue30177) existed in Python itself, but was fixed in version 3.6.2.

## Environment
OS: CentOS Linux release 7.3.1611 (Core)
Python: 3.6.5
pyfakefs: 3.4.1

--- a/pyfakefs/fake_pathlib.py
+++ b/pyfakefs/fake_pathlib.py
@@ -228,8 +228,8 @@ class _FakeFlavour(pathlib._Flavour):
         return parts
 
     def _resolve_posix(self, path, strict):
-        """Original implementation in Python 3.6."""
         sep = self.sep
+        accessor = path._accessor
         seen = {}
 
         def _resolve(path, rest):
@@ -251,27 +251,23 @@ class _FakeFlavour(pathlib._Flavour):
                     if path is not None:
                         # use cached value
                         continue
-                    # The symlink is not resolved, so we must have a
-                    # symlink loop.
+                    # The symlink is not resolved, so we must have a symlink loop.
                     raise RuntimeError("Symlink loop from %r" % newpath)
                 # Resolve the symbolic link
                 try:
                     target = self.filesystem.readlink(newpath)
-                except OSError as exc:
-                    if exc.errno != errno.EINVAL:
-                        if strict:
-                            raise
-                        else:
-                            return newpath
-                    # Not a symlink
+                except OSError as e:
+                    if e.errno != errno.EINVAL and strict:
+                        raise
+                    # Not a symlink, or non-strict mode. We just leave the path
+                    # untouched.
                     path = newpath
                 else:
-                    seen[newpath] = None  # not resolved symlink
+                    seen[newpath] = None # not resolved symlink
                     path = _resolve(path, target)
-                    seen[newpath] = path  # resolved symlink
+                    seen[newpath] = path # resolved symlink
 
             return path
-
         # NOTE: according to POSIX, getcwd() cannot contain path components
         # which are symlinks.
         base = '' if path.is_absolute() else self.filesystem.cwd

--- a/pyfakefs/tests/fake_pathlib_test.py
+++ b/pyfakefs/tests/fake_pathlib_test.py
@@ -406,10 +406,10 @@ class FakePathlibFileObjectPropertyTest(RealPathlibTestCase):
 
     @unittest.skipIf(sys.version_info < (3, 6),
                      'Changed behavior in Python 3.6')
-    def test_resolve_nonexisting_file_before_36(self):
+    def test_resolve_nonexisting_file_after_36(self):
         path = self.path(
             self.make_path('/path', 'to', 'file', 'this can not exist'))
-        self.assertTrue(path, path.resolve())
+        self.assertEqual(path, path.resolve())
         self.assertRaises(FileNotFoundError, path.resolve, strict=True)
 
     def test_cwd(self):

